From 47a647013d9b52f34bdfe1305da28d76e31dde31 Mon Sep 17 00:00:00 2001
From: rncbc <rncbc@rncbc.org>
Date: Fri, 27 Mar 2020 10:07:35 +0000
Subject: [PATCH] - Fixed Setup dialog Cancel (or close) behavior when  
 Settings > Parameters > Frames/Period (aka. buffer-   size) it's the only
 setting that's changed.

Upstream-Status: Applied
---
 CMakeLists.txt            |  2 +-
 ChangeLog                 |  7 +++++++
 configure.ac              |  2 +-
 src/qjackctlSetupForm.cpp | 15 ++++++---------
 src/qjackctlSetupForm.h   |  2 +-
 src/qjackctlSetupForm.ui  |  2 +-
 6 files changed, 17 insertions(+), 13 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 26bd25a..45a6ef7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,7 +2,7 @@ project(QJACKCTL)
 
 cmake_minimum_required(VERSION 3.1)
 
-set (VERSION "0.6.2")
+set (VERSION "0.6.3")
 
 set (CONFIG_VERSION ${VERSION})
 execute_process (
diff --git a/ChangeLog b/ChangeLog
index 3abc93e..39d42cf 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -4,6 +4,13 @@ QjackCtl - JACK Audio Connection Kit Qt GUI Interface
 ChangeLog
 
 
+GIT HEAD
+
+- Fixed Setup dialog Cancel (or close) behavior when
+  Settings > Parameters > Frames/Period (aka. buffer-
+  size) it's the only setting that's changed.
+
+  
 0.6.2  2020-03-24  A Spring'20 Release.
 
 - A scalable (.svg) icon version has been added.
diff --git a/configure.ac b/configure.ac
index a7263e3..faaee8e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1,5 +1,5 @@
 # Process this file with autoconf to produce a configure script.
-AC_INIT(QjackCtl, 0.6.2, rncbc@rncbc.org, qjackctl)
+AC_INIT(QjackCtl, 0.6.3, rncbc@rncbc.org, qjackctl)
 
 AC_CONFIG_SRCDIR(src/qjackctl.cpp)
 AC_CONFIG_HEADERS(src/config.h)
diff --git a/src/qjackctlSetupForm.cpp b/src/qjackctlSetupForm.cpp
index 1493070..44f20dd 100644
--- a/src/qjackctlSetupForm.cpp
+++ b/src/qjackctlSetupForm.cpp
@@ -1,7 +1,7 @@
 // qjackctlSetupForm.cpp
 //
 /****************************************************************************
-   Copyright (C) 2003-2019, rncbc aka Rui Nuno Capela. All rights reserved.
+   Copyright (C) 2003-2020, rncbc aka Rui Nuno Capela. All rights reserved.
 
    This program is free software; you can redistribute it and/or
    modify it under the terms of the GNU General Public License
@@ -1255,9 +1255,9 @@ void qjackctlSetupForm::stabilizeForm (void)
 
 	changeDriverUpdate(m_ui.DriverComboBox->currentText(), false);
 
-	bValid = (bValid || m_iDirtyBuffSize > 0);
-	m_ui.DialogButtonBox->button(QDialogButtonBox::Apply)->setEnabled(bValid);
-	m_ui.DialogButtonBox->button(QDialogButtonBox::Ok)->setEnabled(bValid);
+	bEnabled = (bValid || m_iDirtyBuffSize > 0);
+	m_ui.DialogButtonBox->button(QDialogButtonBox::Apply)->setEnabled(bEnabled);
+	m_ui.DialogButtonBox->button(QDialogButtonBox::Ok)->setEnabled(bEnabled);
 }
 
 
@@ -1908,7 +1908,7 @@ bool qjackctlSetupForm::queryClose (void)
 	bool bQueryClose = true;
 
 	// Check if there's any pending changes...
-	if (m_iDirtySettings > 0 || m_iDirtyOptions > 0) {
+	if (m_iDirtySettings > 0 || m_iDirtyOptions > 0 || m_iDirtyBuffSize > 0) {
 		switch (QMessageBox::warning(isVisible() ? this : parentWidget(),
 			tr("Warning") + " - " QJACKCTL_SUBTITLE1,
 			tr("Some settings have been changed.\n\n"
@@ -1917,12 +1917,9 @@ bool qjackctlSetupForm::queryClose (void)
 			QMessageBox::Discard |
 			QMessageBox::Cancel)) {
 		case QMessageBox::Apply:
-			accept();
+			apply();
 			// Fall thru...
 		case QMessageBox::Discard:
-			// Reset dirty flags...
-			m_iDirtySettings = 0;
-			m_iDirtyOptions = 0;
 			break;
 		default:    // Cancel.
 			bQueryClose = false;
diff --git a/src/qjackctlSetupForm.h b/src/qjackctlSetupForm.h
index fa0dcdd..d35de59 100644
--- a/src/qjackctlSetupForm.h
+++ b/src/qjackctlSetupForm.h
@@ -1,7 +1,7 @@
 // qjackctlSetupForm.h
 //
 /****************************************************************************
-   Copyright (C) 2003-2019, rncbc aka Rui Nuno Capela. All rights reserved.
+   Copyright (C) 2003-2020, rncbc aka Rui Nuno Capela. All rights reserved.
 
    This program is free software; you can redistribute it and/or
    modify it under the terms of the GNU General Public License
diff --git a/src/qjackctlSetupForm.ui b/src/qjackctlSetupForm.ui
index d8465c2..a8051fa 100644
--- a/src/qjackctlSetupForm.ui
+++ b/src/qjackctlSetupForm.ui
@@ -3,7 +3,7 @@
  <author>rncbc aka Rui Nuno Capela</author>
  <comment>JACK Audio Connection Kit - Qt GUI Interface.
 
-   Copyright (C) 2003-2019, rncbc aka Rui Nuno Capela. All rights reserved.
+   Copyright (C) 2003-2020, rncbc aka Rui Nuno Capela. All rights reserved.
 
    This program is free software; you can redistribute it and/or
    modify it under the terms of the GNU General Public License
-- 
2.21.1

